assets/src/amp-story-editor-blocks.js

```diff
@@ -36,7 +58,29 @@ domReady( () => {
 } );
 
 subscribe( () => {
-	setDefaultBlockName( getSelectedBlockClientId() ? 'amp/amp-story-text' : 'amp/amp-story-page' );
+	const defaultBlockName = getDefaultBlockName();
+	const selectedBlockClientId = getSelectedBlockClientId();
+
+	// Switch default block depending on context
+	if ( selectedBlockClientId ) {
+		const selectedBlock = getBlock( selectedBlockClientId );
+
+		if ( 'amp/amp-story-page' === selectedBlock.name && 'amp/amp-story-page' !== defaultBlockName ) {
+			setDefaultBlockName( 'amp/amp-story-page' );
+		} else if ( 'amp/amp-story-page' !== selectedBlock.name && 'amp/amp-story-text' !== defaultBlockName ) {
+			setDefaultBlockName( 'amp/amp-story-text' );
+		}
+	} else if ( ! selectedBlockClientId && 'amp/amp-story-page' !== defaultBlockName ) {
+		setDefaultBlockName( 'amp/amp-story-page' );
+	}
+} );
+
+store.subscribe( () => {
+	Object.keys( getAnimationOrder() ).map( ( page ) => {
```

Comment:
Per above, why no `for...of`?

```js
for ( const page of Object.keys( getAnimationOrder() ).map( ( page ) => getBlock( page ) ).filter( Boolean ) ) {
    removePage( store.getState(), page );
}
```

Reply:
I like the suggestion, but I'd go all the way with the chaining. I think it is more readable that way:

```js
Object.keys( getAnimationOrder() )
    .map( ( page ) => getBlock( page ) )
    .filter( Boolean )
    .map( ( page ) => removePage( store.getState(), page ) );
```